### DISCUSSION TITLE
Help to filter data

### DISCUSSION THREAD

**Discussion by OlegCinema** [prisma/prisma#7632] [Q&A] [answered]
Hello. I have a database with this schema: 
```prisma
model VerificationRequest {
  id         Int      @id @default(autoincrement())
  identifier String
  url  String
  token      String   @unique
  expires    DateTime
  createdAt  DateTime @default(now()) @map(name: "created_at")
  updatedAt  DateTime @default(now()) @map(name: "updated_at")

  @@map(name: "verification_requests")
}
```

1. How can I find all records that was created 23 January for example...? 
2. I have "url". How can I find all urls where it's started with "GitHub.com/..." and has "https" or "http" protocols? 

thanks

**Comment by ryands17** [comment, marked as answer]
@OlegCinema 👋 

1.

```ts
  const start = new Date(2021, 0, 23)
  const end = new Date(2021, 0, 24)
  const result1 = await prisma.verificationRequest.findMany({
    where: { AND: [{ createdAt: { gte: start } }, { createdAt: { lt: end } }] },
  })
```

2.

```ts
const result2 = await prisma.verificationRequest.findMany({
    where: {
      AND: [
        { url: { contains: 'github.com' } },
        {
          OR: [
            { url: { startsWith: 'http' } },
            { url: { startsWith: 'https' } },
          ],
        },
      ],
    },
  })
```

You can find all the operators [here](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference).

**Reply by OlegCinema** [reply]
I would like to ask about something... I have query par in request (to server) equal (it can be true or false; It will respond with equal url if it's true and contains if it's false). How to write code better without duplicated elements? Thanks

**Reply by ryands17** [reply]
Add a condition in your object like this:

```
where: {
  field: {
    condition ? { equals: 'value' } : { contains: 'value }
  }
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
